api-server/server.js

```diff
@@ -124,6 +140,42 @@ app.post("/authenticate", (req, res) => {
   });
 });
 
+
+/******************************* Rooms ********************************/
+
+app.get("/rooms", passport.authenticate(["jwt"], { session: false }), (req, res) => {
+  let userLatitude = req.user.latitude;
+  let userLongitude = req.user.longitude;
+
+  let nearbyRoomPromise = Room.query().select(
+    raw("*, (6371 * acos(cos(radians(:userLatitude)) * " + 
+                        "cos(radians(latitude)) * " + 
+                        "cos(radians(longitude) - " + 
+                            "radians(:userLongitude)) + " +
+                        "sin(radians(:userLatitude)) * " +
+                        "sin(radians(latitude)))) as distance", {
+    userLatitude: userLatitude,
+    userLongitude: userLongitude
+  })).having("distance", "<=", 1).orderBy("distance", "asc");
+
+  nearbyRoomPromise.then((rooms) => {
+    res.json({
+      rooms: rooms
+    });
+  });
+});
+
+app.get("/subscriptions", passport.authenticate(["jwt"], { session: false }), (req, res) => {
```

Comment:
subscriptions are rooms that a users are in ah?

Reply:
Yup. Might delete it with migrations later. See first. Lol.